sdk/eidos_sdk/cpu/agent_cpu.py

```diff
@@ -69,25 +71,31 @@ def __init__(self, call_context: CallContext, cpu: AgentCPU):
         self._cpu = cpu
 
     async def set_boot_messages(
-        self,
-        prompts: List[CPUMessageTypes],
-        output_format: Union[Literal["str"], Dict[str, Any]] = "str",
+            self,
+            prompts: List[CPUMessageTypes],
+            output_format: Union[Literal["str"], Dict[str, Any]] = "str",
     ):
         return await self._cpu.set_boot_messages(self._call_context, list(prompts), output_format)
 
     async def schedule_request(
-        self,
-        prompts: List[CPUMessageTypes],
-        output_format: Union[Literal["str"], Dict[str, Any], type] = "str",
-    ) -> Any:
-        if isinstance(output_format, type):
-            if issubclass(output_format, BaseModel):
-                rtn = await self._cpu.schedule_request(self._call_context, prompts, output_format.model_json_schema())
-                return output_format.model_validate(rtn)
-            else:
-                raise ValueError("type output_format must be a pydantic BaseModel")
-        else:
-            return await self._cpu.schedule_request(self._call_context, prompts, output_format)
+            self,
+            prompts: List[CPUMessageTypes],
+            output_format: T = str,
+    ) -> T:
+        if get_origin(output_format) == list:
+            class Wrapper(BaseModel):
+                items: output_format
+
+            rtn = await self.schedule_request(prompts, Wrapper)
+            return rtn.items
+
+        model = TypeAdapter(output_format)
+        schema = model.json_schema()
```

Comment:
I think we need to do a type check here in case we have an actual schema dict, then we can't convert it to a TypeAdapter before hand

Reply:
we can do it via "get_python_type" from you schema_to_model module.

In this scenario we probably still want to validate, but return the raw python object